app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -147,7 +148,7 @@ open class HomeActivity : AppCompatActivity() {
         // Toggle off the open_link_in_private_tab pref if we are no longer set as the default browser
         // We do this on a separate thread to alleviate performance issues
         val weakReferenceContext = WeakReference(this)
-        lifecycleScope.launch {
+        lifecycleScope.launch(Dispatchers.IO) {
```

Comment:
Every time I see something become asynchronous, I ask: is there a simpler way to do this? There are two things I can think of in this case:
1. Does this need to happen here, or at all?
1. If yes, can we make Browsers faster?

I looked into 1 and I don't think it has to happen here: can you look into whether or not an alternative implementation is worthwhile? Given this is 20ms (on my Pixel 2), I'm wondering if it's worth our time to come up with a new, simpler implementation.